diff --git a/src/config/types.ts b/src/config/types.ts
index 8084be186..8ba783fcb 100644
--- a/src/config/types.ts
+++ b/src/config/types.ts
@@ -60,6 +60,8 @@ export type Source = {
 	mediaProxy?: string;
 
 	signToActivityPubGet?: boolean;
+
+	filesPath?: string;
 };
 
 /**
diff --git a/src/services/drive/internal-storage.ts b/src/services/drive/internal-storage.ts
index deaf3dc83..8a3bf9bc0 100644
--- a/src/services/drive/internal-storage.ts
+++ b/src/services/drive/internal-storage.ts
@@ -9,7 +9,7 @@ const _filename = __filename;
 const _dirname = dirname(_filename);
 
 export class InternalStorage {
-	private static readonly path = Path.resolve(_dirname, '../../../files');
+	private static readonly path = config.filesPath || Path.resolve(_dirname, '../../../files');
 
 	public static resolvePath = (key: string) => Path.resolve(InternalStorage.path, key);
 
